### PR TITLE
Add negative test for input binding.

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Added negative test for input binding.

Will add a decision record PR separately.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1809

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/runtime_test.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ const (
 	TestRuntimeConfigID  = "consumer0"
 	TestPubsubName       = "testpubsub"
 	TestSecondPubsubName = "testpubsub2"
+	testDataString       = "test"
```
Comment (mukundansundar) [inline pkg/runtime/runtime_test.go:50]: changed to fix linter error.

File: tests/apps/binding_input/app.go [hunk 2]
```diff
@@ -18,8 +18,10 @@ import (
 const appPort = 3000
 
 type messageBuffer struct {
-	lock     *sync.RWMutex
-	messages []string
+	lock          *sync.RWMutex
+	messages      []string
+	errorOnce     bool
```
Comment (youngbupark) [inline tests/apps/binding_input/app.go:23]: is there the reason why we need to get only one error?

Comment (mukundansundar) [reply tests/apps/binding_input/app.go:23]: Added more clarifying comments.

File: tests/apps/binding_input/app.go [hunk 3]
```diff
@@ -18,8 +18,10 @@ import (
 const appPort = 3000
 
 type messageBuffer struct {
-	lock     *sync.RWMutex
-	messages []string
+	lock          *sync.RWMutex
+	messages      []string
```
Comment (youngbupark) [inline tests/apps/binding_input/app.go:22]: Can you rename this to `successMessages`?

File: tests/apps/binding_input/app.go (hunk 2 above)
Comment (youngbupark) [inline tests/apps/binding_input/app.go:23]: I would add comment for `errorOnce`

File: tests/e2e/bindings/bindings_test.go [hunk 4]
```diff
@@ -96,21 +99,32 @@ func TestBindings(t *testing.T) {
...
-	require.Equal(t, testMessages, decodedResponse.ReceivedMessages)
+
+	// Only the first message fails, all other messages are successfully consumed.
+	// nine messages succeed.
+	require.Equal(t, testMessages[1:], decodedResponse.ReceivedMessages)
+	// one message fails.
+	require.Equal(t, testMessages[0], decodedResponse.FailedMessage)
+}
+
+func postAssertHTTPStatus(t *testing.T, url string, data []byte, status int) []byte {
```
Comment (youngbupark) [inline tests/e2e/bindings/bindings_test.go:125]: `httpPostWithAssert` or similar sounds better than `postAssertHTTPStatus`. what do you think?

**Review by youngbupark — CHANGES_REQUESTED** [review]

File: tests/e2e/bindings/bindings_test.go (hunk 4 above)
Comment (mukundansundar) [reply tests/e2e/bindings/bindings_test.go:125]: it is specifically asserting no error and only HTTP status .... so only went with postAssert`HTTPStatus`

**Comment by mukundansundar** [comment]
E2E tests failed in windows because Dapr was not able to start. Completely independent of this PR. https://github.com/dapr/dapr/runs/1234747210?check_suite_focus=true#step:13:30

**Review by youngbupark — APPROVED** [review]
Thanks for this work!

**Pull request merged**